src/wallet/wallet.cpp

```diff
@@ -1572,38 +1572,49 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlock* pblock)
     MarkAffectedTransactionsDirty(tx);
 }
 
-void CWallet::SyncCertificate(const CScCertificate& cert, const CBlock* pblock)
+void CWallet::SyncCertificate(const CScCertificate& cert, const CBlock* pblock, int bwtMaturityHeight)
 {
     LOCK2(cs_main, cs_wallet);
     if (!AddToWalletIfInvolvingMe(cert, pblock, true))
         return; // Not one of ours
 
     MarkAffectedTransactionsDirty(cert);
```

Comment:
Is it correct to mark it dirty here instead of doing at the end of the method?

Reply:
Indeed MarkAffectedTransactionsDirty only reset the flag related to credit/debit calculation and does not affect `bwtMaturityHeight `nor `areBwtCeased`.
However for readability I'll move `MarkAffectedTransactionsDirty` to the bottom of the method as for `CWallet::SyncTransaction` and `CWallet::SyncBwtCeasing`